sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,238 @@
+    /// Manipulating resource permissions is only supported when ADLS interop is enabled.
+    /// </summary>
+    public class PathPermissions
+    {
+        /// <summary>
+        /// The <see cref="RolePermissions"/> for the owner of the resource.
+        /// </summary>
+        public RolePermissions Owner { get; set; }
+
+        /// <summary>
+        /// The <see cref="RolePermissions"/> for the owning group of the resource.
+        /// </summary>
+        public RolePermissions Group { get; set; }
+
+        /// <summary>
+        /// The <see cref="RolePermissions"/> for the other users.
+        /// </summary>
+        public RolePermissions Other { get; set; }
+
+        /// <summary>
+        /// If the sticky bit has been set. The sticky bit may be set on directories, the files in that
+        /// directory may only be renamed or deleted by the file's owner, the directory's owner, or the root user.
+        /// </summary>
+        public bool StickyBit { get; set; }
+
+        /// <summary>
+        /// Whether or not there is more permissions information in the ACLs. The permissions string only returns
+        /// information on the owner, owning group, and other, but the ACLs may contain more permissions for specific users
+        /// or groups.
+        /// </summary>
+        public bool ExtendedInfoInAcl { get; set; }
+
+        /// <summary>
+        /// Internal empty constructor.
+        /// </summary>
+        public PathPermissions() { }
+
+        /// <summary>
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit,
+            bool extendedInfoInAcl)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octal">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octal)
+        {
+            if (octal == null)
+            {
+                throw Errors.ArgumentNull(nameof(octal));
+            }
+
+            if (octal.Length != 4)
+            {
+                throw Errors.InvalidArgument(nameof(octal));
```

Comment:
This is an unhelpful exception.  You should throw exceptions with custom messages for all of these with the relevant detail to diagnose the problem.

Reply:
Will do, I’m adding better exception messaging in my next iteration.